uniswap/uniswap.py

```diff
@@ -537,11 +584,45 @@ def _token_to_eth_swap_input(
                 ),
             )
         elif self.version == 3:
+            if recipient is None:
+                recipient = self.address
+
             if fee_on_transfer:
                 raise Exception("fee on transfer not supported by Uniswap v3")
-            return self._token_to_token_swap_input(
-                input_token, self.get_weth_address(), qty, recipient, fee, slippage
+
+            output_token = self.get_weth_address()
+            min_tokens_bought = int((1 - slippage) * self._get_token_eth_input_price(input_token, qty, fee=fee))
+            sqrtPriceLimitX96 = 0
+
+            # Prepare swap data
+            swap_arg_types = ["address", "address", "uint24", "address", "uint256", "uint256", "uint256", "uint160"]
+            swap_arg_types_str = f"({','.join(swap_arg_types)})"
+
+            swap_selector = function_signature_to_4byte_selector(f"exactInputSingle({swap_arg_types_str})")
+
+            swap_args = [input_token, output_token, fee, ETH_ADDRESS, self._deadline(), qty, min_tokens_bought, sqrtPriceLimitX96]
+            swap_args_encoded = encode_abi(swap_arg_types, swap_args)
+
+            # Prepare unwrap data
+            unwrap_arg_types = ["uint256", "address"]
+            unwrap_arg_types_str = f"{','.join(unwrap_arg_types)}"
+
+            unwrap_selector = function_signature_to_4byte_selector(f"unwrapWETH9({unwrap_arg_types_str})")
```

Comment:
You can use contract.encodeABI("unwrapWETH9", [eth_amount, recipient]) instead of doing it manually, this way you don't have to specify the arg types, since the contract is already aware of them.